hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheReplicationOperation.java

```diff
@@ -128,10 +128,9 @@ protected void writeInternal(ObjectDataOutput out)
             for (Map.Entry<Data, CacheRecord> e : cacheMap.entrySet()) {
                 final Data key = e.getKey();
                 final CacheRecord record = e.getValue();
-                final long expirationTime = record.getExpirationTime();
 
-                // If entry is already expired we skip it
-                if (expirationTime > now) {
+                // If entry is already expired or it's tombstone, we just skip it
+                if (!record.isExpiredAt(now) && !record.isTombstone()) {
```

Comment:
Consider rewriting to `if (record.isExpired(now) || record.isTombstone()) continue;` Then it nicely matches the comment, which can be deleted as redundant.